src/tools/perf/lib/libperf.c

```diff
@@ -890,48 +890,49 @@ static ucs_status_t ucp_perf_test_fill_params(ucx_perf_params_t *params,
     return UCS_OK;
 }
 
+static ucs_status_ptr_t
+ucp_perf_test_destroy_ep(ucp_ep_h ep, ucp_rkey_h rkey, unsigned index)
+{
+    ucs_status_ptr_t *req;
+
+    if (rkey != NULL) {
+        ucp_rkey_destroy(rkey);
+    }
+
+    if (ep != NULL) {
+        req = ucp_ep_close_nb(ep, UCP_EP_CLOSE_MODE_FLUSH);
+
+        if (UCS_PTR_IS_PTR(req)) {
+            return req;
+        } else if (UCS_PTR_STATUS(req) != UCS_OK) {
+            ucs_warn("failed to close ep %p on thread %d: %s\n", ep, index,
+                     ucs_status_string(UCS_PTR_STATUS(req)));
+        }
+    }
+    return NULL;
+}
+
 static void ucp_perf_test_destroy_eps(ucx_perf_context_t* perf)
 {
     unsigned i, thread_count = perf->params.thread_count;
     unsigned num_in_prog     = 0;
-    ucs_status_ptr_t **reqs  = ucs_alloca(2 * /* (ep + self_ep) */
-                                          thread_count * sizeof(*reqs));
+    ucs_status_ptr_t **reqs  = ucs_alloca(thread_count * sizeof(*reqs));
```

Comment:
I was mistaken, we do need a request also for self_ep. missed that it's a destroy flow and it's a close request.
so need to do `reqs[num_in_prog++] = req;` for it instead of releasing it.